lib/managedservicesplatform/iam/database.go

```diff
@@ -0,0 +1,121 @@
+package iam
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/openfga/openfga/assets"
+	"github.com/pressly/goose/v3"
+	"github.com/redis/go-redis/v9"
+	"github.com/sourcegraph/log"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.uber.org/zap"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	gormlogger "gorm.io/gorm/logger"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/lib/redislock"
+)
+
+const databaseName = "msp-iam"
+
+type metadata struct {
+	AuthorizationModelID  string    `gorm:"column:authorization_model_id"`
+	AuthorizationModelDSL string    `gorm:"column:authorization_model_dsl"`
+	UpdatedAt             time.Time `gorm:"not null"`
+}
+
+// migrateAndRecon migrates the "msp-iam" database schema (when needed) and
+// recons the framework metadata.
+func migrateAndRecon(ctx context.Context, logger log.Logger, sqlDB *sql.DB, redisClient *redis.Client) (_ *metadata, err error) {
+	ctx, span := iamTracer.Start(ctx, "iam.migrateAndRecon")
```

Comment:
Maybe this is just carry-over from my gamer days but "recon" strikes me as "reconnaissance" which I've never seen in a software context... I assume you mean "reconcile"? This is a pretty common term and not that much longer, I suggest we use the full word

```suggestion
// migrateAndReconcile migrates the "msp-iam" database schema (when needed) and
// recons the framework metadata.
func migrateAndReconile(ctx context.Context, logger log.Logger, sqlDB *sql.DB, redisClient *redis.Client) (_ *metadata, err error) {
	ctx, span := iamTracer.Start(ctx, "iam.migrateAndReconcile")
```

Reply:
I do mean "reconnaissance" 😂 "Reconile" works here too I think.

Reply:
ah, or `interrogates`?

if it really is `recon` not a huge deal, but a bit confused myself 😆